src/nvs.rs

```diff
@@ -352,6 +352,270 @@ impl<T: NvsPartitionId> EspNvs<T> {
 
         Ok(true)
     }
+
+    pub fn len_str(&self, name: &str) -> Result<Option<usize>, EspError> {
+        let c_key = CString::new(name).unwrap();
+
+        #[allow(unused_assignments)]
+        let mut len = 0;
+
+        match unsafe { nvs_get_str(self.1, c_key.as_ptr(), ptr::null_mut(), &mut len as *mut _) } {
+            ESP_ERR_NVS_NOT_FOUND => Ok(None),
+            err => {
+                // bail on error
+                esp!(err)?;
+
+                Ok(Some(len))
+            }
+        }
+    }
+
+    pub fn get_str<'a>(&self, name: &str, buf: &'a mut [u8]) -> Result<Option<&'a [u8]>, EspError> {
```

Comment:
Ideally, this should return an `&'a str`, not `&'a [u8]`. You can go from `&'a [u8]` to `&'a str` by using `core::str::from_utf8_unchecked`.

Reply:
By the way... given that the get_XXX / set_XXX methods are *not* trait implementations, and that inside them we are full of allocations (you know, the `CString` stuff), we can just as well surrender to `alloc` and require the `alloc` module for this API. (And not compile this code if the `alloc` feature is not enabled.)

This would allow you to use `String` and `Vec`, rather than suffering with the `&[u8]` bufs.

Maybe food for thought. Not certain myself which way to go.

Reply:
I tend to lean towards an approach which does not require heap allocation. There might be usage scenarios where reading for example config values from the nvs happens often (no app side caching of the values). For such a scenario having heap allocation might not be optimal. I would prefere stack alloc (not to fragment the heap over time too much). What do you think?